subsys/bluetooth/host/CMakeLists.txt

```diff
@@ -1,57 +1,55 @@
+                            IFDEF:${CONFIG_BT_DEBUG_MONITOR}    monitor.c
+                            IFDEF:${CONFIG_BT_TINYCRYPT_ECC}    hci_ecc.c
+                            IFDEF:${CONFIG_BT_A2DP}             a2dp.c
+                            IFDEF:${CONFIG_BT_AVDTP}            avdtp.c
+                            IFDEF:${CONFIG_BT_RFCOMM}           rfcomm.c
+                            IFDEF:${CONFIG_BT_TESTING}          testing.c
+                            IFDEF:${CONFIG_BT_SETTINGS}         settings.c
+                            IFDEF:${CONFIG_BT_BREDR}            keys_br.c
+                                                                l2cap_br.c
+                                                                sdp.c
+                            IFDEF:${CONFIG_BT_HFP_HF}           hfp_hf.c
+                                                                at.c
+                            IFDEF:${CONFIG_BT_BREDR}            keys_br.c
+                                                                l2cap_br.c
+                                                                sdp.c
+)
 
 if(CONFIG_BT_HCI_HOST)
-  zephyr_library_sources(
-    uuid.c
-    hci_core.c
-    )
-  zephyr_library_sources_ifdef(
-    CONFIG_BT_HOST_CRYPTO
-    crypto.c
-    )
+  zephyr_populate_source_list(OUTPUT PRIVATE_SOURCES
+                              PREPEND_PATH ${CMAKE_CURRENT_LIST_DIR}
+                              APPEND
+                              uuid.c
+                              hci_core.c
+                              IFDEF:${CONFIG_BT_HOST_CRYPTO} crypto.c
+                              IFDEF:${CONFIG_BT_CONN}        conn.c
+                                                             l2cap.c
+                                                             att.c
+                                                             gatt.c
+  )
 
   if(CONFIG_BT_CONN)
-    zephyr_library_sources(
-      conn.c
-      l2cap.c
-      att.c
-      gatt.c
-      )
-
-    if(CONFIG_BT_SMP)
-      zephyr_library_sources(
-        smp.c
-        keys.c
-        )
-    else()
-      zephyr_library_sources(
-        smp_null.c
-        )
-    endif()
+    zephyr_populate_source_list(OUTPUT PRIVATE_SOURCES
+                                APPEND
+                                IFDEF:${CONFIG_BT_SMP}  smp.c
+                                                        keys.c
+                                IFNDEF:${CONFIG_BT_SMP} smp_null.c
+    )
   endif()
 endif()
 
+message("${PRIVATE_SOURCES}")
+
+add_library(subsys__bluetooth__host STATIC ${PRIVATE_SOURCES})
+target_link_libraries(subsys__bluetooth__host zephyr_interface subsys__bluetooth )
+
+# The zephyr_append_cmake_library is discussed as part of issue #8441
+# therefore disregards this from the RFC here.
+# Placed temporary in RFC to allow successful linking without while keeping changes at a minimum.
+zephyr_append_cmake_library(subsys__bluetooth__host)
```

Comment:
They should not.
But I wanted to keep the different RFC focused on specific topics, to avoid discussing too many topics in same.
So this RFC shows a suggestion for #8439 and #8451 uses this and then remove the offending line.

I completely agree, this is not a line every developer should write